programmatic_execution_of_notebooks/DemoQCResultsWithParameters.ipynb

```diff
@@ -0,0 +1,1240 @@
+    "\n",
+    "Use the `DATASET` parameter to select which dataset you'd like to examine:\n",
+    "\n",
+    "* `simons` for the Simons Genome Diversity Project\n",
+    "* `thousand_genomes` for the 1000 Genomes Phase 3 Release\n",
+    "* `platinum_genomes` for the Deep Variant platinum genomes\n",
+    "\n",
+    "Use the `CSV_OUTPUT_FILE_NAME` parameter to name your CSV output file. If you leave it unset, it will default to `<DATASET>_problem_summary.csv`.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "BILLING_PROJECT_ID <- Sys.getenv('GOOGLE_PROJECT')\n",
+    "DATASET <- 'simons'\n",
+    "CSV_OUTPUT_FILE_NAME <- NULL"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Validate Papermill parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DATASETS = c('simons', 'thousand_genomes', 'platinum_genomes')\n",
+    "\n",
+    "if (!(DATASET %in% DATASETS)) {\n",
+    "    stop(str_glue(DATASET, ' is not a valid dataset name. Valid names are: [', \n",
+    "                  paste(DATASETS, collapse = ', '), ']'))\n",
+    "}\n",
+    "\n",
+    "if (is.null(CSV_OUTPUT_FILE_NAME)) {\n",
+    "    CSV_OUTPUT_FILE_NAME <- str_glue(DATASET, '_problem_summary.csv')\n",
+    "}\n",
+    "\n",
+    "print(DATASET)\n",
+    "print(CSV_OUTPUT_FILE_NAME)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Retrieve sample information"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "retrieve_simons_sample_info <- function() {\n",
+    "    return(bq_table_download(bq_project_query(BILLING_PROJECT_ID, '\n",
+    "    SELECT\n",
+    "      id_from_vcf AS name,\n",
+    "      \"Illumina\" AS sequencing_platform,\n",
+    "      sex,\n",
+    "      region AS ancestry\n",
+    "    FROM\n",
+    "      `bigquery-public-data.human_genome_variants.simons_genome_diversity_project_sample_attributes`\n",
+    "    ')))\n",
+    "}\n",
+    "\n",
+    "retrieve_thousand_genomes_sample_info <- function() {\n",
+    "    return(bq_table_download(bq_project_query(BILLING_PROJECT_ID, '\n",
+    "    SELECT\n",
+    "      Sample AS name,\n",
+    "      \"Illumina\" AS sequencing_platform,\n",
+    "      Gender AS sex,\n",
+    "      Super_Population AS ancestry\n",
+    "    FROM\n",
+    "      `bigquery-public-data.human_genome_variants.1000_genomes_sample_info`\n",
+    "    ')))\n",
+    "}\n",
+    "\n",
+    "SAMPLE_INFO_FNS <- list(simons = retrieve_simons_sample_info,\n",
```

Comment:
It would be great to move all the constants to a cell towards the top instead of mixing them in with the logic.

All in one cell:
```
CHROM_RESULTS_PATHS <- list(simons = 'gs://genomics-public-data/simons-genome-diversity-project/reports/Simons_Genome_Diversity_Project_sample_reference_results.csv',
                            platinum_genomes = 'gs://genomics-public-data/platinum-genomes/reports/DeepVariant_Platinum_Genomes_sample_reference_results.csv',
                            thousand_genomes = 'gs://genomics-public-data/1000-genomes-phase-3/reports/1000_Genomes_phase_3_sample_reference_results.csv')

GENOME_RESULTS_PATHS <- list(simons = 'gs://genomics-public-data/simons-genome-diversity-project/reports/Simons_Genome_Diversity_Project_sample_results.csv',
                             platinum_genomes = 'gs://genomics-public-data/platinum-genomes/reports/DeepVariant_Platinum_Genomes_sample_results.csv',
                             thousand_genomes = 'gs://genomics-public-data/1000-genomes-phase-3/reports/1000_Genomes_phase_3_sample_results.csv')


simons_sample_info_query <- '
    SELECT
      id_from_vcf AS name,
      "Illumina" AS sequencing_platform,
      sex,
      region AS ancestry
    FROM
      `bigquery-public-data.human_genome_variants.simons_genome_diversity_project_sample_attributes`
    '

thousand_genomes_sample_info_query <- '
    SELECT
      Sample AS name,
      "Illumina" AS sequencing_platform,
      Gender AS sex,
      Super_Population AS ancestry
    FROM
      `bigquery-public-data.human_genome_variants.1000_genomes_sample_info`
    '

SAMPLE_INFO_QUERIES <- list(simons = simons_sample_info_query,
                        thousand_genomes = thousand_genomes_sample_info_query,
                        platinum_genomes = thousand_genomes_sample_info_query) # Use 1000 genomes for DV platinum
```